File: sensa-template-service/src/main/java/com/sensa/templateservice/entity/Template.java
package com.sensa.templateservice.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.proxy.HibernateProxy;

import java.util.Objects;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(
        name = "template",
        uniqueConstraints =  {
                @UniqueConstraint(name = "template_unique_clientUsername-title", columnNames = {"clientUsername", "title"})
        }
)
public class Template {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String clientUsername;

    @Column(nullable = false)
    private String title;

    private String content;

    @Override
    public final boolean equals(Object o) {
        if (this == o) return true;
        if (o == null) return false;
        Class<?> oEffectiveClass = o instanceof HibernateProxy proxy ? proxy.getHibernateLazyInitializer().getPersistentClass() : o.getClass();
        Class<?> thisEffectiveClass = this instanceof HibernateProxy proxy ? proxy.getHibernateLazyInitializer().getPersistentClass() : this.getClass();
        if (thisEffectiveClass != oEffectiveClass) return false;
        Template template = (Template) o;
        return getId() != null && Objects.equals(getId(), template.getId());
    }

    @Override
    public final int hashCode() {
        return this instanceof HibernateProxy proxy ? proxy.getHibernateLazyInitializer().getPersistentClass().hashCode() : getClass().hashCode();
    }

    public Template addUsername(String username) {
        setClientUsername(username);
        return this;
    }
}

File: template-service/src/main/kotlin/com/sensa/templateservice/entity/Template.kt
package com.sensa.templateservice.entity

import jakarta.persistence.*
import java.time.LocalDateTime
import java.util.*

@Entity
@Table(name = "templates")
class Template(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(name = "template_name", nullable = false, unique = true)
    var templateName: String,

    @Column(name = "description", length = 500)
    var description: String? = null,

    @Column(name = "channel", nullable = false, length = 10)
    @Enumerated(EnumType.STRING)
    var channel: TemplateChannel,

    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    var content: String,

    @Column(name = "created_by", nullable = false)
    var createdBy: UUID,

    @Column(name = "created_at", nullable = false)
    var createdAt: LocalDateTime = LocalDateTime.now(),

    @Column(name = "updated_at", nullable = false)
    var updatedAt: LocalDateTime = LocalDateTime.now()
)
